src/dmd/aggregate.d

```diff
@@ -108,8 +108,9 @@ extern (C++) abstract class AggregateDeclaration : ScopeDsymbol
     // it would be stored in TypeInfo_Class.defaultConstructor
     CtorDeclaration defaultCtor;
 
-    Dsymbol aliasthis;      // forward unresolved lookups to aliasthis
-    bool noDefaultCtor;     // no default construction
+    Dsymbol aliasthis;         // forward unresolved lookups to aliasthis
+    bool hasDeprecatedAliasThis;  // true if alias this is deprecated
```

Comment:
It feels quite specific to have a field for this. It's not possible to store in `aliasthis`?

Reply:
It is, but the aggregate declaration is not keeping any information about the alias this AST node; it just stores the aliased symbol

Reply:
So `aliasthis` is the aliased symbol? Perhaps the `alias this` node could be stored instead. It could potentially be used for other things as well. While `hasDeprecatedAliasThis` could not. I think the `alias this` node can be set here:
https://github.com/dlang/dmd/blob/28f6c6dbc61060f7957d74ab7d2707d8a59b9277/src/dmd/dsymbolsem.d#L540

Reply:
@RazvanN7 : Yeah those destructive AST pass will come back to bite us at one point or another.
And it generally makes DMD as a library much less useful.
So if we can keep the info intact, that'd be much cleaner IMO.